src/cli/global/install.rs

```diff
@@ -151,20 +185,24 @@ fn is_executable(prefix: &Prefix, relative_path: &Path) -> bool {
 }
 
 /// Create the executable scripts by modifying the activation script
-/// to activate the environment and run the executable
-async fn create_executable_scripts(
+/// to activate the environment and run the executable.
+///
+/// If `dry_run` is true, return the filenames of the scripts that would be created but don't
+/// actually write them to disk.
+pub(crate) async fn create_executable_scripts(
```

Comment:
Instead of a dry run I would prefer if you split this function in two, one that creates the content and one that writes the content.